cmd/manager/main.go

```diff
@@ -117,14 +117,9 @@ func main() {
 		os.Exit(1)
 	}
 
-	// Setup all Controllers
-	if err := controller.AddToManager(mgr); err != nil {
-		log.Error(err, "")
-		os.Exit(1)
-	}
-
+	localmetrics.Collector = localmetrics.NewMetricsCollector()
 	metricsServer := metrics.NewBuilder().WithPort(metricsPort).WithPath(metricsPath).
-		WithCollectors(localmetrics.MetricsList).
+		WithCollector(localmetrics.Collector).
```

Comment:
A Collector is a prometheus type that all different type of metrics implement. The object passed here has a collectors list within it, but `WithCollector` is just expecting one metric

Reply:
Right, so I implemented the `Collector` interface. Is there a problem with this approach?

Reply:
Would all future metrics, even non-reconcile or api related go in this collector?

Reply:
AFAICT this is the pattern aws-account-operator uses.

https://github.com/openshift/aws-account-operator/blob/4bb28f759e24fdc03ccf943c9ec772150a421cd4/cmd/manager/main.go#L136
https://github.com/openshift/aws-account-operator/blob/4bb28f759e24fdc03ccf943c9ec772150a421cd4/pkg/localmetrics/localmetrics.go#L37-L55

seems fine by me.

Reply:
huh, I didn't see that in the refactor. AFAIK, that's a pattern for proxying metrics from other monitoring stores and exporters, not for instrumenting metrics. But whatever, if it works